### DISCUSSION TITLE
tutorials/voyage-part3/07-publishing-and-deployment

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#104] [course feedback]
# tutorials/voyage-part3/07-publishing-and-deployment

https://www.apollographql.com/tutorials/voyage-part3/07-publishing-and-deployment

**Comment by jlongo-coforge** [comment]
```
# It runs after the PR is closed with a merge
on:
pull_request:
  types: [closed]
```

This is not a good because a pull request closed will also trigger the staging deployment. You should use "on push" here.

**Reply by torressam333** [reply]
I kind of agree with you...but can you elaborate a bit on why closed would not work as well as push?
Too bad pull_request doesn't have a "merged" event b/c then we would know that it was reviewed by peers and explicitly chosen to be merged and thus is inherently "finalized" and ready to be deployed.
